src/draw_target.rs

```diff
@@ -551,11 +556,48 @@ impl DrawState {
         self.lines.clear();
         self.orphan_lines_count = 0;
     }
+
+    pub(crate) fn visual_line_count<I: SliceIndex<[String], Output = [String]>>(
+        &self,
+        range: I,
+        width: usize,
+    ) -> VisualLines {
+        visual_line_count(&self.lines[range], width)
+    }
+}
+
+#[derive(Add, AddAssign, Clone, Copy, Debug, Default, Eq, Ord, PartialEq, PartialOrd, Sub)]
+pub(crate) struct VisualLines(usize);
+
+impl<T: Into<usize>> From<T> for VisualLines {
+    fn from(value: T) -> Self {
+        Self(value.into())
+    }
+}
+
+impl VisualLines {
+    pub(crate) fn as_usize(&self) -> usize {
+        self.0
+    }
+    pub(crate) fn saturating_add(&self, other: Self) -> Self {
+        Self(self.0.saturating_add(other.0))
+    }
+    pub(crate) fn saturating_sub(&self, other: Self) -> Self {
+        Self(self.0.saturating_sub(other.0))
+    }
+    pub(crate) fn zero() -> Self {
+        Self(0)
+    }
```

Comment:
Style nits: let's order these `zero()` (if we're keeping it), the `saturating` methods, then `as_usize()`, each separated by an empty line.